(DOCSP-29016): Added SEO guidelines. (#144)
mongodb/docs-meta: source/style-guide/seo-guidelines.txt

.. _seo-guidelines:

=====================================
Search Engine Optimization Guidelines
=====================================

.. default-domain:: mongodb

Search Engine Optimization (SEO) considers the actual terms that users 
enter into serach engines (keywords) and employs best practices to 
improve traffic to web pages from search engines. The following 
guidance should help you optimize your documentation pages for search.

Titles
------

Title pages and subsections according to the following SEO best 
practices:

- Use a maximum of 70 characters.

> - Include target keywords.

Reviewer: Suggestion:
It wasn't that clear to me what "target keywords" meant in the context of Titles. 
Does this mean "use words relevant to the topic of the section"? Or does it have something to do with HREF targets? Would appreciate further detail in this bullet point.

- Avoid excessive or irrelevant words (keyword stuffing).

> - Use a unique title for every page.

Reviewer: Questions: 
Would this guidance apply to pages that might share titles in different sets of documentation? For example, "find a document" could be a title of a page in Server, Drivers, mongosh, etc. 
Also, if this applies to subsections, then should we avoid using section titles like "Example", "Syntax", and "Further Learning"?

Author: This mostly concerns titles at the page level even between doc sets. Identical titles might compete in search results.

  
To learn more, see :ref:`titles-and-headings`.

Alternative Text
----------------

> Screen readers read alternative text aloud so that users can better
> understand an on-page image. Specify alterative text according to the 

Reviewer: Suggestion:
I think it could be good to clarify that alternative text only applies to images to avoid confusion. Also found a typo.

```suggestion
Screen readers read alternative text for images aloud so that visually impaired 
users can better understand what the images show. Specify alternative text according to the 
```

Author: I prefer to keep the user definition general to account for any edge cases where someone might use a screen reader beyond visual impairment.

following SEO best practices:

- Use a maximum of 125 characters per image.

> - Describe the image with specific language.

Reviewer: Question:
I wasn't sure what "specific language" represents. Since there isn't another item that explains what it should contain, maybe this should be something like "describe the image with sufficient detail to understand what it shows without seeing it."

- If it makes sense, include keywords.

To learn more, see :ref:`write-for-accessibility`.

Keywords
--------

You can add the actual terms that users enter into search engines 
(keywords) to your page to improve its SEO. Add keywords according to 

> the following best practices:

Reviewer: Suggestion:

I think it might be better to suggest that keywords may not help SEO nor be used to match up with search terms, at least with Google. See
https://developers.google.com/search/docs/crawling-indexing/special-tags#unsupported
for more info.

Author: Yes, there's two parts to these recommendations. We should use the language that users search for in our content. But, search engines might not account for the keyword tags that we use.

- Take the most concise form of the information that the page conveys 
  and make that the target keyword.

- Don't add keywords so broad that they compete with the product page.

- Don't add keywords so specific that we miss the actual behavior of 
  our searchers. For example, add “mongodb atlas course” instead of 
  “Getting Started with Atlas Data Platform”.

To learn more, see :ref:`metadata`.

Descriptions
------------

> The snippet that appears under the link in the search results can 
> impact SEO. Write these descriptions according to the following best 

Reviewer: Suggestion:

Define the term introduced by the title in the first sentence.

```suggestion
The description is a snippet that appears under the link in the search results 
which can impact SEO. Write descriptions according to the following best 
```

practices:

- Make the descriptions informative (and enticing if possible).

- Emphasize the “why” for using the page.

- Use a maximum of 155 characters.

> - Include target keywords and a call to action.

Reviewer: Question: 
I thought Call to Action (CTA) is usually a UI element such as a button that contains text that prompts a user to do do something such as click. I don't think that's something you can include in a text tag. If it's something else, could that be explained here?

Author: We want to call the user to action with the description appears in the search result.

Reviewer: Not a blocker for me, but maybe including an example could be helpful. Does this mean including text like "Click here to learn more" in the meta description?

- Keep the description short, simple, and concise. 

> - Concisely, explain the page content.

Reviewer: Suggestion:
These two items seem similar, and perhaps related to the first item. Maybe they could all be combined, e.g.:

"Use a concise description of the page content that is enticing, if possible."

> - Use unique meta descriptions for every page.

Reviewer: Suggestion:

I think it's probably better to stay consistent in the terminology ("descriptions" vs "meta descriptions"). It currently seems like "descriptions" is the term used more often, so it could be good to use that here too:

```suggestion
- Use unique descriptions for every page.
```

.. _seo-guidelines:

=====================================
Search Engine Optimization Guidelines
=====================================

Search Engine Optimization (SEO) considers the actual terms that users 
enter into search engines (keywords) and employs best practices to 
improve traffic to web pages from search engines. The following 
guidance should help you optimize your documentation pages for search.

Titles
------

> Title pages and subsections according to the following SEO best 

Reviewer 2: [suggestion, wording]

It wasn't immediately clear to me that we're using "title" as a verb in this sentence. On initial reading I thought "title page" was a noun. Perhaps we can tweak this to something like:

"Use the following SEO best practices for page and subsection titles: "

practices:

- Use a maximum of 70 characters.

- Include target keywords (the actual terms that users 
  enter into search engines to reach your page).

- Avoid excessive or irrelevant words (keyword stuffing).

- Strive for unique page titles. Identical titles, even between 
  documentation sets, might compete in search results.
  
To learn more, see :ref:`titles-and-headings`.

Alternative Text
----------------

Screen readers read alternative text for images aloud so that users can 
better understand an image. Specify alternative text according to the 
following SEO best practices:

- Use a maximum of 125 characters per image.

- Describe the image with sufficient detail to understand what it shows.

- If it makes sense, include keywords (the actual terms that users 

>   enter into search engines).

Reviewer 2: [question]

The phrase "if it makes sense" makes me wonder in which situations this practice _doesn't_ make sense. Is that something we can possibly clarify a bit here?

Author: I guess it's a question about whether the keywords for the page really apply to the image. I can think of examples where they might not. I'll change the wording to "If they apply to the image..."

To learn more, see :ref:`write-for-accessibility`.

Keywords
--------

You can add the actual terms that users enter into search engines 
(keywords) to your page to improve its SEO. Add keywords according to 
the following best practices:

- Take the most concise form of the information that the page conveys 

>   and make that the target keyword.

Reviewer 2: [suggestion, non-blocking]

Would it be possible to include an example of this practice? For instance, I would find it helpful if we provided a link to some docs page and highlighted what the target keyword would be for that page.
